proxy/create_container_interceptor.go

```diff
@@ -76,48 +83,114 @@ func (i *createContainerInterceptor) InterceptRequest(r *http.Request) error {
 	return nil
 }
 
-func (i *createContainerInterceptor) addWeaveWaitVolume(hostConfig *docker.HostConfig) {
+func (i *createContainerInterceptor) containerHostname(r *http.Request, container map[string]interface{}) (hostname string, err error) {
+	hostname = r.URL.Query().Get("name")
+	if i.proxy.Config.HostnameFromLabel != "" {
+		hostname, err = i.hostnameFromLabel(hostname, container)
+	}
+	hostname = i.proxy.hostnameMatchRegexp.ReplaceAllString(hostname, i.proxy.HostnameReplacement)
+	return
+}
+
+func (i *createContainerInterceptor) hostnameFromLabel(hostname string, container map[string]interface{}) (string, error) {
+	labels, err := lookupObject(container, "Labels")
+	if err != nil {
+		return "", err
+	}
+
+	labelIface, ok := labels[i.proxy.Config.HostnameFromLabel]
```

Comment:
From here to the end of the function is pretty much lookupString() - could you just call that?